### ISSUE TITLE
server: TestAdminAPITableStats stressrace-flaky due to context timeout

### ISSUE THREAD

**Issue by tschottdorf** [cockroachdb/cockroach#8890]
```
--- FAIL: TestAdminAPITableStats (4.86s)
	admin_cluster_test.go:120: status: 408 Request Timeout, body: {
		  "Error": "context deadline exceeded",
		  "Code": 4
		}
FAIL
```

**Comment by tschottdorf** [comment]
New failure:

```
make stressrace PKG=./server TESTS=TestAdminAPITableStats

--- FAIL: TestAdminAPITableStats (10.67s)
	admin_cluster_test.go:78: pq: context deadline exceeded
FAIL


ERROR: exit status 1

17 runs completed, 1 failures, over 11s

```

Looks like a 10s timeout is set on some context.

**Comment by tamird** [comment]
For posterity: 201f32b closed the original incarnation of this test, but the issue now refers to a new (currently unfixed) failure.

**Comment by mrtracy** [comment]
Minimal reproduce is this:

```go
func TestAdminAPICreateDatabase(t *testing.T) {
	defer leaktest.AfterTest(t)()

	const nodeCount = 3
	tc := testcluster.StartTestCluster(t, nodeCount, base.TestClusterArgs{
		ReplicationMode: base.ReplicationAuto,
		ServerArgs: base.TestServerArgs{
			ScanInterval:    time.Millisecond,
			ScanMaxIdleTime: time.Millisecond,
		},
	})
	defer tc.Stopper().Stop()
	if err := tc.WaitForFullReplication(); err != nil {
		t.Fatal(err)
	}

	db := tc.ServerConn(0)
	if _, err := db.Exec(`CREATE DATABASE "test test"`); err != nil {
		t.Fatal(err)
	}
}
```

No Admin API necessary, this fails stressrace basically immediately every time, either with the "context deadline exceeded", or with data races detected:

```

==================
WARNING: DATA RACE
Read at 0x00c42155f980 by goroutine 112:
  reflect.maplen()
      /usr/local/go/src/runtime/hashmap.go:1044 +0x0
  reflect.Value.MapKeys()
      /usr/local/go/src/reflect/value.go:1072 +0x427
  encoding/gob.(*Encoder).encodeMap()
      /usr/local/go/src/encoding/gob/encode.go:377 +0x225
  encoding/gob.encOpFor.func3()
      /usr/local/go/src/encoding/gob/encode.go:577 +0x158
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.encOpFor.func4()
      /usr/local/go/src/encoding/gob/encode.go:587 +0x116
  encoding/gob.(*Encoder).encodeArray()
      /usr/local/go/src/encoding/gob/encode.go:357 +0x2c9
  encoding/gob.encOpFor.func1()
      /usr/local/go/src/encoding/gob/encode.go:557 +0x170
  encoding/gob.(*Encoder).encodeStruct()
      /usr/local/go/src/encoding/gob/encode.go:334 +0x437
  encoding/gob.(*Encoder).encode()
      /usr/local/go/src/encoding/gob/encode.go:707 +0x257
  encoding/gob.(*Encoder).EncodeValue()
      /usr/local/go/src/encoding/gob/encoder.go:247 +0x6ea
  encoding/gob.(*Encoder).Encode()
      /usr/local/go/src/encoding/gob/encoder.go:175 +0x72
  github.com/cockroachdb/cockroach/kv.grpcTransportFactory.func1()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/transport_race.go:94 +0x39b
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:190 +0x8a

Previous write at 0x00c42155f980 by goroutine 179:
  runtime.mapassign1()
      /usr/local/go/src/runtime/hashmap.go:442 +0x0
  github.com/cockroachdb/cockroach/storage.(*Store).Send.func1()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/store.go:1965 +0x690
  github.com/cockroachdb/cockroach/storage.(*Store).Send()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/store.go:2051 +0x10a3
  github.com/cockroachdb/cockroach/storage.(*Stores).Send()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/stores.go:182 +0x24a
  github.com/cockroachdb/cockroach/server.(*Node).Batch.func3()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/server/node.go:823 +0x3d3
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunTask()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:218 +0x10d
  github.com/cockroachdb/cockroach/server.(*Node).Batch()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/server/node.go:835 +0x2ce
  github.com/cockroachdb/cockroach/roachpb._Internal_Batch_Handler()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/roachpb/api.pb.go:1530 +0x345
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /Users/matt/go/src/google.golang.org/grpc/server.go:611 +0x10b6
  google.golang.org/grpc.(*Server).handleStream()
      /Users/matt/go/src/google.golang.org/grpc/server.go:769 +0xf9c
  google.golang.org/grpc.(*Server).serveStreams.func1.1()
      /Users/matt/go/src/google.golang.org/grpc/server.go:422 +0xb8

Goroutine 112 (running) created at:
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:191 +0x73
  github.com/cockroachdb/cockroach/kv.grpcTransportFactory()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/transport_race.go:100 +0x2d8
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendToReplicas()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:998 +0x157
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendRPC()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:379 +0x27b
  github.com/cockroachdb/cockroach/kv.(*DistSender).RangeLookup()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:265 +0x30e
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).performRangeLookup()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:459 +0x4da
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:344 +0x677
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:282 +0xbf
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).performRangeLookup()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:455 +0x6c8
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).lookupRangeDescriptorInternal()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:344 +0x677
  github.com/cockroachdb/cockroach/kv.(*rangeDescriptorCache).LookupRangeDescriptor()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/range_cache.go:282 +0xbf
  github.com/cockroachdb/cockroach/kv.(*DistSender).getDescriptors()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:444 +0x101
  github.com/cockroachdb/cockroach/kv.(*DistSender).sendChunk()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:689 +0x3eb
  github.com/cockroachdb/cockroach/kv.(*DistSender).Send()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/dist_sender.go:606 +0x1a0
  github.com/cockroachdb/cockroach/kv.(*TxnCoordSender).Send()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/kv/txn_coord_sender.go:391 +0x689
  github.com/cockroachdb/cockroach/internal/client.(*DB).send()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/internal/client/db.go:495 +0x293
  github.com/cockroachdb/cockroach/internal/client.(*DB).(github.com/cockroachdb/cockroach/internal/client.send)-fm()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/internal/client/db.go:440 +0x7c
  github.com/cockroachdb/cockroach/internal/client.sendAndFill()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/internal/client/db.go:422 +0x1ab
  github.com/cockroachdb/cockroach/internal/client.(*DB).Run()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/internal/client/db.go:440 +0xd4
  github.com/cockroachdb/cockroach/internal/client.runOneResult()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/internal/client/db.go:522 +0x97
  github.com/cockroachdb/cockroach/internal/client.(*DB).AdminSplit()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/internal/client/db.go:373 +0xdc
  github.com/cockroachdb/cockroach/storage.(*splitQueue).process()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/split_queue.go:101 +0x35b
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processReplica()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/queue.go:489 +0x64b
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1.2()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/queue.go:428 +0x6f
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunTask()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:218 +0x10d
  github.com/cockroachdb/cockroach/storage.(*baseQueue).processLoop.func1()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/storage/queue.go:432 +0x359
  github.com/cockroachdb/cockroach/util/stop.(*Stopper).RunWorker.func1()
      /Users/matt/go/src/github.com/cockroachdb/cockroach/util/stop/stopper.go:190 +0x8a

Goroutine 179 (finished) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /Users/matt/go/src/google.golang.org/grpc/server.go:423 +0xba
  google.golang.org/grpc/transport.(*http2Server).operateHeaders()
      /Users/matt/go/src/google.golang.org/grpc/transport/http2_server.go:221 +0x126d
  google.golang.org/grpc/transport.(*http2Server).HandleStreams()
      /Users/matt/go/src/google.golang.org/grpc/transport/http2_server.go:282 +0x7f3
  google.golang.org/grpc.(*Server).serveStreams()
      /Users/matt/go/src/google.golang.org/grpc/server.go:424 +0x1d5
  google.golang.org/grpc.(*Server).serveNewHTTP2Transport()
      /Users/matt/go/src/google.golang.org/grpc/server.go:411 +0x4f0
  google.golang.org/grpc.(*Server).handleRawConn()
      /Users/matt/go/src/google.golang.org/grpc/server.go:388 +0x5c2
==================
```

**Comment by mrtracy** [comment]
This never fails in regular stress, it fails almost immediately in stressrace, which makes me inclined to believe this is a problem with the race-specific code.

**Comment by tschottdorf** [comment]
I also get these stressing your mini test  above:

```
from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.608287 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.620369 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.714757 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.757394 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.876695 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.930365 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:13.945842 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.031446 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.039708 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.090813 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.112806 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.141066 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.164686 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.200813 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.299810 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.331219 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.348509 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.427419 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.464469 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.521940 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.576046 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.625124 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.638288 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.686337 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.704553 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
E160907 13:49:14.711351 184 storage/queue.go:517  [replicate] on [n1,s1,r1{/Min-/Table/11}]: change replicas of range 1 failed: reservation declined:header:<node_id:3 store_id:3 > from_node_id:1 from_store_id:1 range_id:1 range_size:400302
panic: test timed out after 2m0s

goroutine 5275 [running]:
panic(0x1949e00, 0xc421bc5210)
       	/home/tschottdorf/go/src/runtime/panic.go:500 +0x1ae
testing.startAlarm.func1()
       	/home/tschottdorf/go/src/testing/testing.go:918 +0x14e
created by time.goFunc
       	/home/tschottdorf/go/src/time/sleep.go:154 +0x78

goroutine 1 [chan receive, 1 minutes]:
testing.(*T).Run(0xc4200b2fc0, 0x1aedee9, 0x1a, 0x1d639e8, 0x77a701)
       	/home/tschottdorf/go/src/testing/testing.go:647 +0x56e
testing.RunTests.func1(0xc4200b2fc0)
       	/home/tschottdorf/go/src/testing/testing.go:793 +0xba
testing.tRunner(0xc4200b2fc0, 0xc420105d78)
       	/home/tschottdorf/go/src/testing/testing.go:610 +0xca
testing.RunTests(0x1d65678, 0x25e0220, 0x36, 0x36, 0x0)
       	/home/tschottdorf/go/src/testing/testing.go:799 +0x4bb
testing.(*M).Run(0xc420105ef0, 0x32ec8c0)
       	/home/tschottdorf/go/src/testing/testing.go:743 +0x130
github.com/cockroachdb/cockroach/server_test.TestMain(0xc420105ef0)
       	/home/tschottdorf/src/github.com/cockroachdb/cockroach/server/main_test.go:32 +0xd7
main.main()
       	github.com/cockroachdb/cockroach/server/_test/_testmain.go:160 +0x1b6

goroutine 17 [syscall, 1 minutes, locked to thread]:
runtime.goexit()
       	/home/tschottdorf/go/src/runtime/asm_amd64.s:2086 +0x1

goroutine 34 [chan receive]:
github.com/cockroachdb/cockroach/util/log.(*loggingT).flushDaemon(0x292d100)
       	/home/tschottdorf/src/github.com/cockroachdb/cockroach/util/log/clog.go:1032 +0x85
created by github.com/cockroachdb/cockroach/util/log.init.1
       	/home/tschottdorf/src/github.com/cockroachdb/cockroach/util/log/clog.go:613 +0xc3

goroutine 8 [select]:
github.com/cockroachdb/cockroach/util/retry.(*Retry).Next(0xc422187d60, 0xc422187d20)
       	/home/tschottdorf/src/github.com/cockroachdb/cockroach/util/retry/retry.go:128 +0x307
github.com/cockroachdb/cockroach/testutils/testcluster.(*TestCluster).WaitForFullReplication(0xc420188060, 0x1d647d8, 0xc4201fe000)
       	/home/tschottdorf/src/github.com/cockroachdb/cockroach/testutils/testcluster/testcluster.go:475 +0x132
github.com/cockroachdb/cockroach/server_test.TestAdminAPICreateDatabase(0xc4200b3080)
       	/home/tschottdorf/src/github.com/cockroachdb/cockroach/server/admin_cluster_test.go:158 +0x136
testing.tRunner(0xc4200b3080, 0x1d639e8)
       	/home/tschottdorf/go/src/testing/testing.go:610 +0xca
created by testing.(*T).Run
       	/home/tschottdorf/go/src/testing/testing.go:646 +0x530
```

**Comment by tschottdorf** [comment]
I wasn't able to reproduce the data race reliably enough to make quick progress on this. However, there is reason to believe that the race isn't completely trivial to find - in the transport we're cloning the transaction before passing the request down to the local server, so it's unlikely that this path was taken (though I wasn't even able to verify that; the race stopped showing up). However, the other path goes through the GRPC client code, and it seems unlikely (?) that that would bypass the actual networking layer.

Since the race happens in `storage` and the race detection is in `kv`, the request (or parts of it?) must be passing through the transport. Also, the race itself is always when the Store is updating the observed timestamp map, and it looks like `.Clone()` correctly makes a copy.

Help here would be appreciated, I won't get to this for at least a few days.

**Comment by tschottdorf** [comment]
```diff
diff --git a/storage/replica.go b/storage/replica.go
index eab982a..8e8153d 100644
--- a/storage/replica.go
+++ b/storage/replica.go
@@ -1559,6 +1559,10 @@ func (r *Replica) addWriteCmd(
                                } else {
                                        log.Warningf(ctx, "unable to cancel expired Raft command %s", ba)
                                }
+                       case <-r.store.stopper.ShouldQuiesce():
+                               if tryAbandon() {
+                                       pErr = roachpb.NewError(&roachpb.NodeUnavailableError{})
+                               }
                        }
                }
        } else {
```

removes the case which hangs (@BramGruneir when is that hitting master?)
I've started again to try to repro the data race.

**Comment by BramGruneir** [comment]
@tschottdorf Should be ready go.  I'm waiting on a final LGTM. See #9182.

**Comment by tschottdorf** [comment]
@BramGruneir cool, mind adding "Fixes #8890" in that PR?
